### DISCUSSION TITLE
Request for Additional AWS Resource Support, Tests, Append Only Tracking, Cost Estimation, and Report Exporting

### DISCUSSION THREAD

**Discussion by pachico** [tailwarden/komiser#453] [General]
Hello!
First of all, I would like to congratulate you on the Komiser project. It has been extremely helpful in meeting the specific internal policies of the company I work for. However, I have noticed that not all the AWS resources we need to cover are currently supported, such as AutoScaling Groups. Therefore, I have started looking into the source code to implement it and submit a PR.

Additionally, I have noticed that the application lacks tests. I would be happy to contribute to the project by creating some AWS service client interface to do some mocking and using mockery for that purpose. I believe that agreeing on a homogeneous testing strategy would make it easier to add more tests.

Furthermore, I have noticed that there is no "DELETE" operation in the SQLite database, which means that if users want a "what's running now" report, there should be some kind of filter like "where fetched_at >= now() - resource fetching frequency". Is this assumption correct?

I have also noticed that the prices are hardcoded, which is a great way to have an estimation but does not solve price fluctuations or differences in prices between regions. Is there any plan to have a more dynamic source of truth to get prices from?

Lastly, do you have any plan to implement a view to CSV report for exporting the reports? While it is easy to create ad-hoc reports in any format knowing the DB structure, a built-in view to CSV report would be helpful.

Again, I would like to express my appreciation for the project. The current architecture makes it very easy to add additional services. Well done!

Thank you.

**Comment by mlabouardy** [comment]
Hey @pachico 

Really appreciate the kind words :)

You're right about the coverage of the resources, we're supporting new AWS resources (same for other providers) on a weekly basis (thanks to our internal team and contributors), and we have a bunch of [good first issues](https://github.com/tailwarden/komiser/issues?q=is%3Aissue+is%3Aopen+aws) for that, feel free to grab one and would be happy to walk you through the code base, you can join us in our [Discord server](http://discord.tailwarden.com/) if you're not there already where you can meet the team and other contributors.

💯 about adding tests to the project, we're[ currently adding tests](https://github.com/tailwarden/komiser/issues/374) for the frontend/dashboard and planning to include tests for the engine as well. Let's collaborate on this together.

Regarding the DELETE operation, you're right about the assumption since Komiser stores the history but we're going to ship an improvement soon so users can see what's running now by default and use a date picker to go through the past if needed.

Same for pricing, we're working on it to take into account all different parameters including the region (AWS pricing is a nightmare lol)

For the CSV report, we're going to ship it in the [upcoming days](https://roadmap.tailwarden.com/komiser/p/csv-export). In a related topic, last week we shipped Slack alerts, would love to have your [feedback on the feature](https://www.tailwarden.com/changelog) and how we can improve it :) 

Again, thanks a million for the support and the feedback, it will help us in improving the project :)

**Reply by pachico** [reply]
Awesome!
Thanks for your words and quick reply.
I have forked the repo today but I don't expect to have anything ready to share in less than 10 days since next week will be tremendously busy.

However, right after that, I will find myself available for frequent collaboration. Can't wait to start!

Cheers!
